### PR TITLE
AWS - Connect - Create new Connect resource and instance-attribute filter

### PR DESCRIPTION
**Pull request by peterheo**
Addresses issue #7471 

Create new Connect resource for custodian and implement an instance-attribute filter

### CODE REVIEW

File: c7n/resources/connect.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+from c7n.manager import resources
+from c7n.query import QueryResourceManager, TypeInfo
+from c7n.filters import ValueFilter
+from c7n.utils import local_session, type_schema
+
+
+@resources.register('connect')
```
Comment (kapilt) [inline c7n/resources/connect.py:7]: worth a rename to connect-instance afaics

File: c7n/resources/connect.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+class Connect(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'connect'
+        enum_spec = ('list_instances', 'InstanceSummaryList', None)
+        arn_type = 'connect'
+        name = id = 'InstanceId'
+        universal_taggable = object()
+        cfn_type = 'AWS::Connect'
```
Comment (kapilt) [inline c7n/resources/connect.py:16]: afaics connect instances aren't supported by cloudformation per https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/AWS_Connect.html

in which case we can leave this blank.

File: c7n/resources/connect.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+@resources.register('connect-instance')
+class Connect(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'connect'
+        enum_spec = ('list_instances', 'InstanceSummaryList', None)
+        arn_type = 'connect'
+        name = id = 'InstanceId'
+        universal_taggable = object()
```
Comment (kapilt) [inline c7n/resources/connect.py:15]: afaics this resource/service isn't supported by universal tagging, so this field should be omitted.

File: c7n/resources/connect.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+from c7n.utils import local_session, type_schema
+
+
+@resources.register('connect-instance')
+class Connect(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'connect'
+        enum_spec = ('list_instances', 'InstanceSummaryList', None)
+        arn_type = 'connect'
```
Comment (kapilt) [inline c7n/resources/connect.py:13]: looking at the arn of the instance in the recorded test data, the arn type for this type should be 'instance'

**Comment by PratMis** [comment]
looks good to me, @kapilt do you mind taking a look at this one again please?

**Review by kapilt — APPROVED** [review]
lgtm, thanks

**Pull request merged**